core/src/main/java/org/apache/carbondata/core/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.geo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.carbondata.common.exceptions.sql.MalformedCarbonCommandException;
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.util.CustomIndex;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * GeoHash default implementation
+ */
+public class GeoHashImpl extends CustomIndex<List<Long[]>> {
```

Comment:
Can we move all the geohash implementation to new module and new jar

Reply:
I feel geohash is a specific implementation, not supposed to be inside core. Only interfaces should be defined here and implementations can be in another package.

Reply:
Moved `GeoHashImpl`to new module named carbondata-geo. It extends `CustomIndex` .
`CustomIndex` abstract class is in carbondata-core module. And `GeoHashImpl` is instantiated from integration layer(`carbondata-spark-common`) after processing indexHandler property.
So now, dependencies are as follows:
`carbondata-geo` depends on `carbondata-core` and `carbondata-common`
`carbondata-spark-common` depends on `carbondata-geo`

And in `carbondata-spark2` module, a new package `org.apache.carbondata.geo` is added for files - `GeoUtils` and `InPolygonUDF`
And in `carbondata-spark-common-test` module, a new package `org.apache.carbondata.geo` is added for file `GeoTest`. It is a sql based test file for geo.